tests/test_request_attrs.py

```diff
@@ -143,6 +143,23 @@ def test_nonlatin_path(self, test_path):
 
         assert req.path == falcon.uri.decode(test_path)
 
+    @pytest.mark.skipif(not six.PY2, reason='Test only applies to Python 2')
+    @pytest.mark.parametrize('test_path', [
+        u'/hello_\u043f\u0440\u0438\u0432\u0435\u0442',
+        u'/world_\u043c\u0438\u0440',
+        u'/test_\u0442\u0435\u0441\u0442',
+    ])
+    def test_utf_8_path(self, test_path):
+        # NOTE(kandziu): Test that under python 2.6/7 a path with
```

Comment:
You could change this to mention ``2.7`` only since we dropped support for ``2.6``.

Reply:
Could we expand these tests for your fix?

I haven't checked in detail myself, but could one craft a contrived test case where Unicode character expansion in UTF-8 includes the byte ``/`` as part of multiple-byte representation, which cause routing to a wrong end point or something like that?